packages/platform-sdk-profiles/src/environment/env.ts

```diff
@@ -164,11 +159,11 @@ export class Environment {
 			throw new Error("Please call [verify] before booting the environment.");
 		}
 
-		if (this.storage.data) {
+		if (this.storage.data && Object.keys(this.storage.data).length > 0) {
 			this.data().fill(this.storage.data);
 		}
 
-		if (this.storage.profiles) {
+		if (this.storage.profiles && Object.keys(this.storage.profiles).length > 0) {
```

Comment:
@faustbrian added additional condition here as `storage.data` and `storage.profiles` are always set as objects from `verify`

Reply:
The condition before the `&&` is probably not needed then since it will always be defined.